### PR TITLE
feat(sdk-coin-polygon): support recovery

### PR DESCRIPTION
**Pull request by mullapudipruthvik**
Ticket: BG-56221

Add support for Non-BitGo Recoveries

Changes:
recover and signFinal methods

### CODE REVIEW

**Review by byhow — COMMENTED** [review]
Overall LGTM; It'd be good to get someone from the polygon team to fill in reviews for coin-specific context

File: modules/sdk-coin-polygon/src/polygon.ts [hunk 1]
```diff
@@ -306,4 +320,186 @@ export class Polygon extends Eth {
 
     return { halfSigned: txParams };
   }
+
+  /**
+   * Make a query to Etherscan for information such as balance, token balance, solidity calls
```
Comment (gianchandania) [inline modules/sdk-coin-polygon/src/polygon.ts:325]: nit: Make a query to Polygonscan

File: modules/sdk-coin-polygon/src/polygon.ts [hunk 2]
```diff
@@ -306,4 +320,186 @@ export class Polygon extends Eth {
...
+   * @param query {Object} key-value pairs of parameters to append after /api
+   * @returns {Object} response from Etherscan
```
Comment (gianchandania) [inline modules/sdk-coin-polygon/src/polygon.ts:327]: nit: {Promise<Object>} response from Polygonscan

File: modules/sdk-coin-polygon/src/polygon.ts [hunk 3]
```diff
@@ -306,4 +320,186 @@ export class Polygon extends Eth {
 
     return { halfSigned: txParams };
   }
+
+  /**
+   * Make a query to Etherscan for information such as balance, token balance, solidity calls
+   * @param query {Object} key-value pairs of parameters to append after /api
```
Comment (gianchandania) [inline modules/sdk-coin-polygon/src/polygon.ts:326]: nit: @param {Object} query — key-value pairs of parameters to append after /api

File: modules/sdk-coin-polygon/src/polygon.ts [hunk 4]
```diff
@@ -306,4 +320,186 @@ export class Polygon extends Eth {
...
+
+  /**
+   * Builds a funds recovery transaction without BitGo
+   * @param params
+   * @param params.userKey {String} [encrypted] xprv or xpub
+   * @param params.backupKey {String} [encrypted] xprv or xpub if the xprv is held by a KRS provider
+   * @param params.walletPassphrase {String} used to decrypt userKey and backupKey
+   * @param params.walletContractAddress {String} the Polygon address of the wallet contract
+   * @param params.krsProvider {String} necessary if backup key is held by KRS
+   * @param params.recoveryDestination {String} target address to send recovered funds to
```
Comment (gianchandania) [inline modules/sdk-coin-polygon/src/polygon.ts:354]: Please follow the format @param {String} params.userKey - [encrypted] xprv or xpub

File: modules/sdk-core/src/bitgo/environments.ts [hunk 5]
```diff
@@ -99,6 +103,8 @@ const testnetBase: EnvironmentTemplate = {
   etherscanApiToken: process.env.ETHERSCAN_API_TOKEN,
   eth2ExplorerBaseUrl: 'https://beaconscan.com/api',
   ethwExplorerBaseUrl: '', // TODO BG-55611: Update this URL once a ETHW Explorer is announced
+  polygonBaseUrl: 'https://api-testnet.polygonscan.com',
```
[the remark this answers is not in the document]
Comment (mullapudipruthvik) [reply modules/sdk-core/src/bitgo/environments.ts]: its not required to add in mock, by default it contains testnet values

File: modules/sdk-coin-polygon/src/polygon.ts (hunk 2 above)
Comment (mullapudipruthvik) [reply modules/sdk-coin-polygon/src/polygon.ts]: done

**Review by gianchandania — APPROVED** [review]
lgtm

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**